internal/mode/static/state/dataplane/configuration.go

```diff
@@ -637,6 +628,23 @@ func buildTelemetry(g *graph.Graph) Telemetry {
 		tel.Interval = string(*telemetry.Exporter.Interval)
 	}
 
+	ratioMap := make(map[string]int32)
+	for _, pol := range g.NGFPolicies {
```

Comment:
Is it possible to find a way to not be aware of any specific NGF policies here?  I believe the goal of the design of NGF policy framework is to keep everything contained in policies package (cc @kate-osborn )

For example, each observability policy can generate some config that can go to into http context (required for split clients)

Reply:
Not saying it's not possible, just would require a bit of work I think.

Reply:
My thoughts originally when building this were "we have a telemetry template that sets telemetry config at the http context already, and we have a list of policies that could contain configuration for that context, so add it there".

Reply:
I think this approach would need:
- another Generate function for the interface, so we have one for http template and one for server/location template (can't reuse the existing one because it needs to be called for different templates)
- Ideally I would want to build this into the existing telemetry struct because it's directly related, but that still wouldn't be generalized; so we would need a new struct built in the `dataplane.Configuration`, and then a new template to build with that data.

Reply:
Just continuing some thoughts...I feel like we're going to continue to have some special cases. For example, our planned UpstreamSettings policy will only apply to upstreams, and our other policies won't. So I suspect that we're not going to have generalized logic there, it's going to be specifically for building an upstream config from that policy.

Reply:
I see the problem now. This policy attaches to routes but generates config for the http and location contexts.

I don't think the UpstreamSettings policy will have a similar issue because it doesn't generate config for multiple contexts. We just need to attach the policy to the right place in the graph (likely the backend).

I'm ok with leaving this as-is for now and revisiting it after the release. I think there's a way we can make it work, but it's going to take more design time.

Reply:
Let's add a FIXME?